src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -291,4 +301,22 @@ static class Blob {
       this.blobStats = blobStats;
     }
   }
+
+  /**
+   * Get the metadata size (footer size) from a parquet file
+   *
+   * @param bytes the serialized parquet file
+   * @return the extended metadata size
+   */
+  static long getExtendedMetadataSize(byte[] bytes) throws IOException {
+    final int magicOffset = bytes.length - ParquetFileWriter.MAGIC.length;
+    final int footerSizeOffset = magicOffset - Integer.BYTES;
+    if (footerSizeOffset < 0
+        || !ParquetFileWriter.MAGIC_STR.equals(
```

Comment:
curious why is this MAGIC_STR comparison being added only now / why is it needed only when calculating ext metadta size?

Reply:
Addressing this [comment](https://github.com/snowflakedb/snowflake-ingest-java/pull/851#discussion_r1786732455)